docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Dependency injection with the Azure SDK for .NET
+description: Learn how to use dependency injection with the Azure SDK for .NET client libraries
+ms.date: 05/12/2021
+ms.custom: devx-track-dotnet
+ms.author: pakrym
+author: pakrym
+---
+# Dependency injection with the Azure SDK for .NET
+
+This article demonstrates how to register Azure service clients from the [latest .NET Azure SDKs](https://azure.github.io/azure-sdk/releases/latest/index.html) in an ASP.NET Core application. Every ASP.NET Core application starts by booting up the application using the instructions provided in the `Startup` class. This includes a `ConfigureServices` method that is an ideal place to configure clients.
+
+To configure the service clients, first add the following NuGet packages to your project:
+
+- [Microsoft.Extensions.Azure](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/extensions/Microsoft.Extensions.Azure/README.md)
+- [Azure.Identity](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/README.md)
+- The `Azure.*` package you'd like to use.
+
+The sample code in this article will use Key Vault secrets and Blob Storage for demonstration purposes.
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.Azure
+dotnet add package Azure.Identity
+dotnet add package Azure.Security.KeyVault.Secrets
+dotnet add package Azure.Storage.Blobs
+```
+
+## Register client
+
+In the `ConfigureServices` method, register a client for each service:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    // Add a KeyVault client
+    builder.AddSecretClient(keyVaultUrl);
+
+    // Add a Storage account client
+    builder.AddBlobServiceClient(storageUrl);
+
+    // Use DefaultAzureCredential by default
+    builder.UseCredential(new DefaultAzureCredential());
+  });
+
+  services.AddControllers();
+}
+```
+
+In code above, you need to explicitly specify the `keyVaultUrl` and `storageUrl`. Both variables are `Uri` types. The [Store configuration separately from code](#store-configuration-separately-from-code) section shows how you can avoid specifying the urls explicitly.
+
+The code above uses `DefaultAzureCredential` for authentication. [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential) will choose the best authentication mechanism based on your environment, allowing you to move your app seamlessly from development to production with no code changes.
+
+## Use the registered clients
+
+With the clients registered in `Startup`, you can now use them:
+
+```csharp
+[ApiController]
+[Route("[controller]")]
+public class MyApiController : ControllerBase
+{
+  private readonly BlobServiceClient blobServiceClient;
```

Comment:
```suggestion
  private readonly BlobServiceClient _blobServiceClient;
```

Reply:
fixed in the latest commit that adds **See also** section